### PR TITLE
chore: sentry init

### PR DESCRIPTION
**Pull request by Lanttcat**
## Description

Closes #MF-2597
## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: patches/@sentry__core@7.3.0.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+diff --git a/esm/integrations/functiontostring.js b/esm/integrations/functiontostring.js
+index 63309b276b5765bf0feb16ad6ddbe3d763697175..a48fc98a2c00db375fccb39067ed6e5a84e155c9 100644
+--- a/esm/integrations/functiontostring.js
++++ b/esm/integrations/functiontostring.js
+@@ -20,10 +20,12 @@ class FunctionToString  {constructor() { FunctionToString.prototype.__init.call(
+    setupOnce() {
+         originalFunctionToString = Function.prototype.toString;
+ 
+-        Function.prototype.toString = function ( ...args) {
```
Comment (Jack-Works) [inline patches/@sentry__core@7.3.0.patch:9]: please open an PR at the upstream and remove this patch after the PR is merged.

File: packages/shared-base/src/sentry/index.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private _plugin_id?: string
+    private _user?: {
+        persona?: PersonaIdentifier
+        profile?: ProfileIdentifier
+    }
+
+    constructor(
+        platform: string,
+        pluginId?: string,
+        user?: { persona?: PersonaIdentifier; profile?: ProfileIdentifier },
```
Comment (Jack-Works) [inline packages/shared-base/src/sentry/index.ts]: Do we need this information in the logging?

File: packages/scripts/src/extension/normal.ts [hunk 3]
```diff
@@ -26,7 +27,9 @@ export const buildBaseExtension: TaskFunction = buildExtensionFlag('default', {
 })
 
 export async function extensionWatch(f: Function | BuildFlagsExtended) {
+    buildSentry
```
Comment (Jack-Works) [inline packages/scripts/src/extension/normal.ts]:
```suggestion
```

File: packages/mask/src/social-network/ui.ts [hunk 4]
```diff
@@ -114,6 +118,9 @@ export async function activateSocialNetworkUIInner(ui_deferred: SocialNetworkUI.
         if (!(ref.avatar && ref.identifier)) return
         Services.Identity.updateProfileInfo(ref.identifier, { avatarURL: ref.avatar, nickname: ref.nickname })
         const currentProfile = getCurrentIdentifier()
+        SocialNetworkLogs.user = {
+            profile: currentProfile?.identifier,
```
Comment (Jack-Works) [inline packages/mask/src/social-network/ui.ts]: privacy concern?

File: packages/shared-base/src/sentry/index.ts (hunk 2 above)
Comment (Lanttcat) [reply packages/shared-base/src/sentry/index.ts]: Don't need. I will remove it.

File: packages/shared-base/src/sentry/index.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import type { PersonaIdentifier, ProfileIdentifier } from '../Identifier/index.js'
+
+// @ts-ignore
+const Sentry = globalThis.Sentry as Sentry
+
+Sentry.init({
```
Comment (Jack-Works) [inline packages/shared-base/src/sentry/index.ts]: only init it when user choose to opt-in the logging

Comment (Lanttcat) [reply packages/shared-base/src/sentry/index.ts]: yep. I will add opt-in in next commit.

File: packages/mask/shared/legacy-settings/settings.ts [hunk 6]
```diff
@@ -45,6 +45,8 @@ export const dismissVerifyNextID = createComplexNetworkSettings(
 export const bioDescription = createNetworkSettings('bioDescription', '')
 export const personalHomepage = createNetworkSettings('personalHomepage', '')
 
+export const enableLog = createGlobalSettings('enableLog', false)
```
Comment (guanbinrui) [inline packages/mask/shared/legacy-settings/settings.ts]:
```suggestion
export const logSettings = createGlobalSettings('log', false)
```

File: packages/shared/src/sentry/component.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+export const Logger = ({ children, value }: Logger) => {
+    const { logger } = useLoggerContext()
+    const handleClick = useCallback(
+        (e: React.MouseEvent) => {
+            if (!isValidElement(children)) return
+            logger?.captureMessage(value)
+            children.props.onClick?.(e)
+        },
+        [value],
```
Comment (guanbinrui) [inline packages/shared/src/sentry/component.tsx]:
```suggestion
        [value, children, logger],
```

File: packages/shared/src/sentry/base.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        Sentry.captureException(error, scope)
+    }
+
+    public captureMessage(message: string | object): void {
+        const scope = this.initScope()
+        Sentry.captureMessage(message, scope)
+    }
+}
+
+export enum LogsType {
```
Comment (guanbinrui) [inline packages/shared/src/sentry/base.ts]:
```suggestion
export enum LogType {
```

File: packages/shared/src/sentry/base.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            release: process.env.VERSION,
+            tracesSampleRate: 1.0,
+        })
+    }
+
+    set platform(platform: LogPlatform) {
+        this._platform = platform
+    }
+
+    setLogUser(user: { persona?: PersonaIdentifier; profile?: ProfileIdentifier }) {
```
Comment (guanbinrui) [inline packages/shared/src/sentry/base.ts]: Not used?

File: packages/dashboard/src/initialization/Dashboard.tsx [hunk 10]
```diff
@@ -37,6 +40,8 @@ export default function DashboardRoot() {
     }
     const theme = themes[appearance]
 
+    // useEffect(() => logger.captureMessage(LogsType.DashboardAccess), [])
```
Comment (guanbinrui) [inline packages/dashboard/src/initialization/Dashboard.tsx]:
```suggestion
```

File: packages/web3-providers/src/Sentry/index.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+export * from './base.js'
+export * from './context.js'
+export * from './component.js'
+export * from './hook.js'
```
Comment (guanbinrui) [inline packages/web3-providers/src/Sentry/index.ts]: Move to `packages/Sentry`.

**Review by guanbinrui — APPROVED** [review]

**Comment by Jack-Works** [comment]
oh no, not reviewed yet!!

**Pull request merged**